examples/uefi/libvirt.tf

```diff
@@ -0,0 +1,36 @@
+provider "libvirt" {
+    uri = "qemu:///system"
+}
+
+resource "libvirt_volume" "ubuntu-cloud-uefi" {
+  name = "ubuntu-cloud-uefi"
+  source = "https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-uefi1.img"
+}
+
+resource "libvirt_volume" "volume" {
+  name = "vm${count.index}"
+  base_volume_id = "${libvirt_volume.ubuntu-cloud-uefi.id}"
+  count = 1
+}
+
+resource "libvirt_domain" "domain" {
+  count = 2
+  name = "ubuntu-cloud-${count.index}"
+  memory = "512"
+  vcpu = 1
+  # This is the file which will back the UEFI NVRAM content.
+  nvram = "/var/lib/libvirt/qemu/nvram/vm${count.index}_VARS.fd"
+  # This file needs to be provided by the user.
+  nvram_template = "/srv/provisioning/terraform/debian-stable-uefi_VARS.fd"
```

Comment:
Updating the documentation page is actually a very good idea, I'm a bit disappointed I forgot about this. :) 

I also like the idea of encapsulating this under the nvram block, honestly this is how I tried to use the nvram block the first time only to find out it was not a block. Definitely gonna do this.

Finally, you are right, the template is a binary file and it really doesn't sense to template its contents in my point of view. I wish it was possible, for it would be much more convenient when provisioning these UEFI VMs.